### PR TITLE
Updating policy

### PR DESCRIPTION
**Pull request by lyie28**
-> tutorial added update_policy.ipynb
-> code added. Appears to be functional but requires cleaning up and testing

### CODE REVIEW

**Review by dhalf — CHANGES_REQUESTED** [review]
Could you move the update policy tutorial into the defining policy tutorial please?

**Review by dhalf — APPROVED** [review]

File: client/src/bastionlab/polars/remote_polars.py [hunk 1]
```diff
@@ -935,6 +939,9 @@ def fetch(self) -> pl.DataFrame:
     def save(self):
         return self._meta._client.persist_df(self._identifier)
 
+    def update_policy(self, new_policy: Policy):
```
Comment (dhalf) [inline client/src/bastionlab/polars/remote_polars.py:942]: This is already inherited

File: server/bastionlab_polars/src/lib.rs [hunk 2]
```diff
@@ -498,4 +498,35 @@ impl PolarsService for BastionLabPolars {
...
+        if owner == true {
+            let req = &request.get_ref();
+            let new_policy: Policy = serde_json::from_str(req.new_policy.as_ref().unwrap()).map_err(|_| Status::unknown("Failed to deserialize policy."))?;
+            //get dataframe for writing
+            let mut dfs = self.dataframes.write().unwrap(); 
+            let artifact = dfs.get_mut(&request.get_ref().identifier);
+            match artifact{
+                Some(artifact) => {
+                    artifact.policy = new_policy;
+                    drop(dfs);
```
Comment (dhalf) [inline server/bastionlab_polars/src/lib.rs:518]: Consider using blocks instead of explicit drops

**Review by dhalf — CHANGES_REQUESTED** [review]

**Comment by ShannonSD** [comment]
Please be aware that if authentication is disabled anyone can change policies making them useless. Maybe the logic should be to refuse alteration of policies when authentication is disabled.

**Pull request closed without merging**